hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalogUtil.java

```diff
@@ -0,0 +1,105 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.hive;
+
+import java.util.List;
+import java.util.Optional;
+import org.apache.hadoop.hive.metastore.IMetaStoreClient;
+import org.apache.hadoop.hive.metastore.TableType;
+import org.apache.hadoop.hive.metastore.api.NoSuchObjectException;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.iceberg.BaseMetastoreTableOperations;
+import org.apache.iceberg.ClientPool;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.exceptions.NoSuchIcebergViewException;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** A utility class to validate Hive Iceberg Table and Views. */
+final class HiveCatalogUtil {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HiveCatalogUtil.class);
+
+  private HiveCatalogUtil() {
+    // empty constructor for utility class
+  }
+
+  static boolean isTableWithTypeExists(
+      ClientPool<IMetaStoreClient, TException> clients,
+      TableIdentifier identifier,
+      TableType tableType) {
+    String database = identifier.namespace().level(0);
+    String tableName = identifier.name();
+
+    try {
+      List<String> tables = clients.run(client -> client.getTables(database, tableName, tableType));
+      return !tables.isEmpty();
+    } catch (TException e) {
+      throw new RuntimeException(
+          "Failed to check table existence " + database + "." + tableName, e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to listTables", e);
+    }
+  }
+
+  static void validateTableIsIcebergView(Table table, String fullName) {
+    String tableType = table.getParameters().get(BaseMetastoreTableOperations.TABLE_TYPE_PROP);
+    NoSuchIcebergViewException.check(
+        table.getTableType().equalsIgnoreCase(TableType.VIRTUAL_VIEW.name())
+            && tableType != null
+            && tableType.equalsIgnoreCase(BaseMetastoreTableOperations.ICEBERG_TABLE_TYPE_VALUE),
+        "Not an iceberg view: %s (type=%s) (tableType=%s)",
+        fullName,
+        tableType,
+        table.getTableType());
+  }
+
+  static void matchAndThrowExistenceTypeException(Table table) {
+    if (table.getTableType().equalsIgnoreCase(TableType.VIRTUAL_VIEW.name())) {
+      throw new AlreadyExistsException(
+          "View already exists: %s.%s", table.getDbName(), table.getTableName());
+    }
+
+    throw new AlreadyExistsException(
+        "Table already exists: %s.%s", table.getDbName(), table.getTableName());
+  }
+
+  static Optional<Table> loadHmsTableIfPresent(
```

Comment:
We sometimes use this method to get table, sometimes don't.
I think we either should generally use, or we should generally avoid using it - as no really complicated logic is hidden here.

At first glance I would remove this method